lib/ssh-tunnel.js

```diff
@@ -21,17 +21,28 @@ SSHTunnel.prototype.listen = function(done) {
     message: 'Create SSH Tunnel',
     pending: true
   });
+  /**
+   * Workaround a bug in `tunnel-ssh` where the error event is not
+   * handled.  This causes an UncaughtException in cases such as
+   * attempting to use a key which is passphrase protected.
+   *
+   * @see INT-1771
+   */
+
+  this._tunnel = createTunnel(this.model.ssh_tunnel_options, function() {
+    debug('tunnel-ssh called callback');
+  });
 
-  this._tunnel = createTunnel(this.model.ssh_tunnel_options, function(err) {
-    if (err) {
-      debug('error setting up tunnel', err);
-      this.emit('status', {
-        message: 'Create SSH Tunnel',
-        error: err
+  this._tunnel.on('netConnection', function(netConnection) {
+    var onError = this._tunnel.emit.bind(this.__tunnel, 'error');
+    netConnection.on('sshStream', function(sshStream) {
+      sshStream.on('error', function(err) {
+        onError(err);
       });
+    });
+  }.bind(this));
 
-      return done(err);
-    }
+  this.__tunnel.on('listening', function() {
```

Comment:
What's the difference between single and double underscore `_tunnel`, `__tunnel`, as seen on this line and a few above/below? Is this a bug?